commit bd0b62465520c0bf5c21435f356160ad88fd6416
Author: Arjan van de Ven <arjan@linux.intel.com>
Date:   Wed Feb 11 17:28:14 2015 -0600

    UBUNTU: SAUCE: Clear Linux: smpboot: reuse timer calibration
    
    NO point recalibrating for known-constant tsc... saves 200ms+ of boot time.
    
    Author:    Arjan van de Ven <arjan@linux.intel.com>
    
    Signed-off-by: Miguel Bernal Marin <miguel.bernal.marin@linux.intel.com>
    Signed-off-by: Tim Gardner <tim.gardner@canonical.com>

diff --git a/arch/x86/kernel/tsc.c b/arch/x86/kernel/tsc.c
index 78b9cb5..bd1af89 100644
--- a/arch/x86/kernel/tsc.c
+++ b/arch/x86/kernel/tsc.c
@@ -1378,6 +1378,9 @@ unsigned long calibrate_delay_is_known(void)
 	if (!tsc_disabled && !cpu_has(&cpu_data(cpu), X86_FEATURE_CONSTANT_TSC))
 		return 0;
 
+	if (cpu != 0)
+		return cpu_data(0).loops_per_jiffy;
+
 	if (!mask)
 		return 0;
 
